src/core/utils/chainProviders.ts

```diff
@@ -20,20 +19,21 @@ const provider = ({
 
   if (!enableMulticallQueue) {
     logger.debug({ chainId }, `Using standard provider`)
-    return new ethers.providers.StaticJsonRpcProvider(url, chainId)
+    return new ethers.JsonRpcProvider(url, chainId, {
+      staticNetwork: Network.from(chainId),
+    })
   }
 
   logger.debug({ chainId }, 'Using multicall queue provider')
 
-  const provider = new ethers.providers.StaticJsonRpcProvider(url, chainId)
+  const provider = new ethers.JsonRpcProvider(url, chainId)
 
   // deployed on 100+ chains at address
   // https://www.multicall3.com/deployments
-  const multicallContract = new Contract(
+  const multicallContract = Multicall__factory.connect(
```

Comment:
Is there any reason why we were using `ethers.Contract` directly here? It's throwing type errors now.

Reply:
It is fine like you proposed, the thinking was MulticallQueue would have one less dependency in case we wanted to separate to own library, but its fine what you have.